src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -81,6 +86,42 @@ export class InputControlProvider {
         });
     }
 
+    private _initializeDynamicValidations() {
+        this._pipelineTemplate.inputs.forEach((inputDes) => {
+
+            var inputControl = this._inputControlsMap.get(inputDes.id);
+
+            if (!!inputControl && !!inputDes.dynamicValidations && inputDes.dynamicValidations.length > 0) {
+                var dataSourceToInputsMap = new Map<DataSource, InputControl[]>();
+                inputDes.dynamicValidations.forEach((validation: InputDynamicValidation) => {
+                    var validationDataSource = DataSourceUtility.getDataSourceById(this._pipelineTemplate.dataSources, validation.dataSourceId);
+                    if (validationDataSource) {
+                        dataSourceToInputsMap.set(validationDataSource, []);
+
+                        let dynamicValidationRequiredInputIds = DataSourceUtility.getDependentInputIdList(validationDataSource.endpointUrlStem);
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.requestBody));
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.resultTemplate));
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.resultSelector));
+
+                        dynamicValidationRequiredInputIds = Array.from(new Set(dynamicValidationRequiredInputIds));
```

Comment:
why do we need this Array.from ?

Reply:
because dynamicValidationRequiredInputIds is of Array type.

Reply:
But `dynamicValidationRequiredInputIds` is also an array. No?

Reply:
here first a 'Set' is created to get only 'unique values' and then it is converted back to Array.